### PR TITLE
testifylint: enable require-error

### PR DESCRIPTION
**Pull request by mmorel-35**
## Which problem is this PR solving?
- Apply require-error rule of testifylint

## Description of the changes
-  use require instead of assert on errors testing 

## How was this change tested?
- CI 

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: plugin/storage/badger/spanstore/read_write_test.go [hunk 1]
```diff
@@ -607,14 +607,14 @@ func BenchmarkServiceIndexLimitFetch(b *testing.B) {
 
 // Opens a badger db and runs a test on it.
 func runLargeFactoryTest(tb testing.TB, test func(tb testing.TB, sw spanstore.Writer, sr spanstore.Reader)) {
-	assert := assert.New(tb)
+	assertion := require.New(tb)
```
Comment (mmorel-35) [inline plugin/storage/badger/spanstore/read_write_test.go:610]: This is the last other different change . Create require assertions instead of assert

**Review by mmorel-35 — COMMENTED** [review]
Exxept the two quoted changes in this comments, this replacez assert.{NoError,Error,ErrorContains,ErrorIs,EqualError} methods by require.{NoError,Error,ErrorContains,ErrorIs,EqualError}

Would you prefer to split this PR by package of around 30 files maybe ?

File: model/span_test.go [hunk 2]
```diff
@@ -53,21 +53,18 @@ func TestTraceIDMarshalJSONPB(t *testing.T) {
 			ref := model.SpanRef{TraceID: model.NewTraceID(testCase.hi, testCase.lo)}
 			out := new(bytes.Buffer)
 			err := new(jsonpb.Marshaler).Marshal(out, &ref)
-			if assert.NoError(t, err) {
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply model/span_test.go:56]: yeah, in this case it's safe to replace it like this, but it's not always the case.

**Review by yurishkuro — APPROVED** [review]
Thanks!

**Pull request merged**